### ISSUE TITLE
Generate Release Notes not working when using CI

### ISSUE THREAD

**Issue by bredar** [rfennell/vNextBuild#63]
### VSTS Extension you are using
2.0.4 (problem was also on 1.4)
### Expected behavior and actual behavior
After Upgrading to TFS Version 15.105.25910.0 the script does not work anymore if the build is startet per CI. It works if I start the build manually. If its started with CI, i get this error message:
"Issues
One or more of the steps defined in the build does not have an associated task definition."
The error appears before running any build steps. 
All other build actions are working. If I remove or disable just the "Generate Release Notes" task, it works.

### Steps to reproduce the problem
Create and build using CI. 
Make a commit to force a CI build.
-> Problem appears in log.

**Comment by rfennell** [comment]
I think I may have seen the same 'problem' during tests. I think it might be a change in behaviour in VSTS.

If you open the build definition for editing does it show the task, but have a small red flag on the right of task in the workflow?

If you just click on the task the flag disappears, save the definition and you are good to go.

I think the idea is that you have to approve (some) updates

**Comment by bredar** [comment]
Thanks! I do not see any red flags. Do I miss something or do I have an other problem?
![generatereleasenotes](https://cloud.githubusercontent.com/assets/17574488/21713918/aadd99ac-d3fc-11e6-9637-6aef4ce6c6aa.PNG)

**Comment by rfennell** [comment]
So is it still not working, as that all looks fine to me

**Comment by bredar** [comment]
Yes, it does still not work.

Von: Richard Fennell [mailto:notifications@github.com]
Gesendet: Freitag, 6. Januar 2017 13:14
An: rfennell/vNextBuild <vNextBuild@noreply.github.com>
Cc: François Zbinden <zbinden@dataflow-solutions.ch>; Author <author@noreply.github.com>
Betreff: Re: [rfennell/vNextBuild] Generate Release Notes not working when using CI (#63)


So is it still not working, as that all looks fine to me

**Comment by rfennell** [comment]
Can't see why there should be an issue with tfs2017 (TFS Version 15.105.25910.0).

I will regret it on that platform and get back to you.

**Comment by rfennell** [comment]
OK getting confused now. I have installed the 2.0.4 task onto a local TFS 2017 RTM VM and it work exactly as expected, on a private build agent that builds from a Git repo. I see no issues triggering the build manually or as a CI trigger.

Looking at other forums I see a few reported issues of this type of thing [#1170](https://github.com/Microsoft/vsts-tasks/issues/1170) and [#1816]( https://github.com/Microsoft/vsts-tasks/issues/1816). There is mention of clearing caches, reinstalling tasks etc. but nothing concrete.

Best suggestion appears to be to on the TFS Admin console disable SSH support.

Does that help?

**Comment by rfennell** [comment]
Did you make any progress on this?

If I don;t hear anything in 7 days I will close this issue, re-open it if is an issue after that

**Comment by bredar** [comment]
No, I have no solution. For now, I had to disable the release notes.

Von: Richard Fennell [mailto:notifications@github.com]
Gesendet: Mittwoch, 18. Januar 2017 21:39
An: rfennell/vNextBuild <vNextBuild@noreply.github.com>
Cc: François Zbinden <zbinden@dataflow-solutions.ch>; Author <author@noreply.github.com>
Betreff: Re: [rfennell/vNextBuild] Generate Release Notes not working when using CI (#63)


Did you make any progress on this?

If I don;t hear anything in 7 days I will close this issue, re-open it if is an issue after that

**Comment by rfennell** [comment]
So just to confirm
- The release notes work if the build is manually triggered
- But you get this error of the build is triggered via CI

Would it be possible for you to send the the logs of a failing CI build so I can look in detail.

**Comment by stpbl** [comment]
Hi together
I also have same Problem here; manually triggered builds succeed, but the CI-triggered ones not 
(as template I tried with both Default-templates( copies of build-basic-template.md resp. build-dump-template.md )

>> [error]System.Management.Automation.ParameterBindingValidationException: Cannot bind argument to parameter 'InputObject' because it is null.

We use private build-agents with currently V2.4.2 of your build Task here. 

attached a zip-file with debug-enabled log-file and its related template-file
[GenerateReleaseNotes_Build_log.zip](https://github.com/rfennell/vNextBuild/files/747168/GenerateReleaseNotes_Build_log.zip)

**Comment by rfennell** [comment]
Thank you for the detailed report, I will try to repo it.

**Comment by rfennell** [comment]
OK I seem to have a repo, the issue appears to be different objects are passed to the build on a manual and Ci build. specifically for the the $build.parameters is empty on a CI build.

I don' think there is much I can do about the lack of data, but I can fail for gracefully - working on that now

**Comment by rfennell** [comment]
OK go to the bottom of this one. The issue boils down to the fact for a manual build the build parameters are passed, so in a template you can use lines like

````
**Build Config**  : $($build.parameters | ConvertFrom-Json | select -ExpandProperty BuildConfiguration )
```

On a CI build these values are just not there. Nothing I can so about that one, sorry

The problem this missing data causes is that the render function errors. Thus failing the task. I have reworked the render method to trap the error in neater manner, the error if written to the markdown file as opposed to failing the build

This is shipped in 2.5.x
